go/vt/vtgate/engine/set.go

```diff
@@ -395,6 +400,12 @@ func sqlModeChangedValue(qr *sqltypes.Result) (bool, sqltypes.Value) {
 	newValArr := strings.Split(newVal, ",")
 	for _, nVal := range newValArr {
 		nVal = strings.ToUpper(nVal)
+		for _, mode := range unsupportedSQLModes {
+			if mode == nVal {
+				// we are setting an unsupported sql_mode
+				return false, sqltypes.Value{}, vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "setting the %s sql_mode is unsupported", nVal)
+			}
+		}
```

Comment:
I think we should only mark a boolean here and return error only if sql_mode is modified.
```
if changed && unsupportedExists {
 return ..err..
}
```

Reply:
good point